lib/error.g

```diff
@@ -242,7 +243,7 @@ BIND_GLOBAL("ErrorInner",
         res := fail;
     fi;
     ErrorLevel := ErrorLevel-1;
-    ErrorLVars := errorLVars;
+    ErrorLVars := ErrorLVarsOld;
```

Comment:
With this change, nested errors won't work -- i.e. nested break loops.

All you need is the correct `ErrorLVars` in your `OnBreak` handler, right? So you simply leave `errorLVars` as it is, but insert `ErrorLVars := errorLVars;` right before the call to `OnBreak`. That should never cause harms, but would remove the need for your `OnBreak` handler to do anything.

I.e. this PR would essentially insert a single line.

Does that work?

Reply:
Since the nested break loops don't work like this this can't be merged.

I think the problem with setting `ErrorLVars := errorLVars;` right before calling `OnBreak()` is that currently, when entering `OnBreak`, `ErrorLVars` is set to `context`. `Where` uses `ErrorLVars` to access `context` to print the traceback. (if only `OnBreak` could take `context` as an argument..)

Or did I misunderstand something there?

Reply:
OK, I just re-read the whole of `ErrorInner` (as I should have done before my comments *sigh*), and yes, you are right.

But what *should* work is adding a new global `ErrorLVarsOld` as you did, but retaining `errorLVars`; then set `ErrorLVarsOld` only just before calling `OnBreak`. That's not very nice, because we are essentially passing an argument through a global variable, but it'd work.

But yeah, at this point, we might as well introduce `AlwaysShowTracebackOnError` as discussed in issue #2547.

And refactor this code later on again ;-).